src/guidellm/core/result.py

```diff
@@ -148,6 +148,7 @@ def end_time(self) -> float:
         :rtype: float
         """
 
+        self._recording_started()
         assert self._end_time
```

Comment:
So yeah. This assert statement was added to catch your attention 😅

From the philosophical perspective, I remember that we agreed that we were going to use a `pydantic` core for our Entities. So the assertion is just a quick hint for the mypy ( since the GitHub Actions is already there ) and they will probably be removed after you merge your *Serializers* Pull Request.

Reply:
What do you think?
It will be easier for me to resolve conflicts after rebasing

Reply:
Ah gotcha, makes sense

Reply:
Yeah, let me finalize the pydantic diff today and then can rebase on that